File: 14th Mar 2022/q1.py
# 1. Find triplets with zero sum

class Solution:
    #Function to find triplets with zero sum.    
    def findTriplets(self, arr, n):
       arr.sort() #sort array
       
       for idx in range(n):                     # for(int idx = 1; idx <= n; i++)
           start = idx + 1
           end = n - 1
           
           while start < end:
               sum = arr[idx] + arr[start] + arr[end]
               
               if sum == 0:
                   return True
               
               elif sum > 0:
                   end -= 1
               
               else:
                   start += 1
       
       return False 

#{ 
#  Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        n=int(input())
        a=list(map(int,input().strip().split()))
        if(Solution().findTriplets(a,n)):
            print(1)
        else:
            print(0)
# } Driver Code Ends

File: 26th Mar 2022/q1.cpp
// 1. Print the left element

// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends

class Solution{
    public:
    int leftElement(int a[], int n) {
        sort(a, a + n);
        int i, j;
        for(i = 0, j = n-1; i <= j; i++, j--)
        {
            if(a[i] == a[j])
                return a[j];
        }
        return a[j];
    }
};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int a[n], i;
        for (i = 0; i < n; i++) cin >> a[i];
        Solution ob;
        cout << ob.leftElement(a, n) << endl;
    }
}
  // } Driver Code Ends